File: C#/958. Check Completeness of a Binary Tree.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    public bool IsCompleteTree(TreeNode root) {
        Queue<TreeNode> q = new Queue<TreeNode>();
        TreeNode n;
        bool f = false;
        q.Enqueue(root);
        while(q.Count>0){
            n = q.Dequeue();
            if(n == null){
                f = true;
            }
            else{
                if(f == true){
                    return false;
                }
                q.Enqueue(n.left);
                q.Enqueue(n.right);
            }
        }
        return true;
    }
}

File: C++/958. Check Completeness of a Binary Tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    bool isCompleteTree(TreeNode* root) {
        queue<TreeNode*> q;
        TreeNode* n;
        bool f = false;
        q.push(root);
        while(q.size()>0){
            n = q.front();
            q.pop();
            if(n == NULL){
                f = true;
            }
            else{
                if(f == true){
                    return false;
                }
                q.push(n->left);
                q.push(n->right);
            }
        }
        return true;
    }
};
